packages/connectivity/src/scp-cf/destination/destination-service.ts

```diff
@@ -251,6 +246,30 @@ export async function fetchCertificate(
   }
 }
 
+function getTenantFromTokens(token: AuthAndExchangeTokens | string): string {
+  if (typeof token === 'string') {
+    const decoded = decodeJwt(token);
+    if (decoded.zid) {
+      return decoded.zid;
+    }
+    if (decoded.iss) {
+      return parseSubdomain(decoded.iss);
+    }
+    throw new Error('Could not obtain tenant identifier from jwt.');
```

Comment:
[req] I think the `exchangeHeaderJwt` wukk also be used in the bring your own JWT approach. In this case the JWT could not contain the `zid` or `iss` property. So perhaps we restructure this a bit that we do not call the method itself but a helper method which checks the zid and iss and return this or undefinend. Then we do all the cases and see if we find something. In the very end we throw if non of the ways worked.

Reply:
Tried to restructure this, have a look.